src/en/mangamonks/src/eu/kanade/tachiyomi/extension/en/mangamonks/MangaMonks.kt

```diff
@@ -0,0 +1,267 @@
+
+import eu.kanade.tachiyomi.extension.en.mangamonks.MangaMonksHelper.buildApiHeaders
+import eu.kanade.tachiyomi.extension.en.mangamonks.MangaMonksHelper.toDate
+import eu.kanade.tachiyomi.extension.en.mangamonks.MangaMonksHelper.toFormRequestBody
+import eu.kanade.tachiyomi.extension.en.mangamonks.MangaMonksHelper.toStatus
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.MissingFieldException
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+
+class MangaMonks : ParsedHttpSource() {
+
+    override val name = "MangaMonks"
+
+    override val baseUrl = "https://mangamonks.com"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    // popular
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/popular-manga/$page", headers)
+    }
+    override fun popularMangaSelector() = ".main-slide"
+    override fun popularMangaNextPageSelector() = "li:nth-last-child(2) a.page-btn"
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            title = element.selectFirst(".detail a")!!.text()
+            setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+            thumbnail_url = element.select("img").attr("data-src")
+        }
+    }
+
+    // latest
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/latest-releases/$page", headers)
+    }
+
+    override fun latestUpdatesSelector() = ".tab-pane .row .col-12"
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    // search
+    private var searchMode: Boolean = false
+
+    private val searchUrl = "https://mangamonks.com/search/live"
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = filters.let { if (it.isEmpty()) getFilterList() else it }
+        return if (query.isNotEmpty()) {
+            searchMode = true
+            val requestBody = query.toFormRequestBody()
+            val requestHeaders = headersBuilder().buildApiHeaders(requestBody)
+
+            POST(searchUrl, requestHeaders, requestBody)
```

Comment:
why not use baseUrl here?
```suggestion
            POST("$baseUrl/search/live", requestHeaders, requestBody)
```